### ISSUE TITLE
tvtv.us is returning 403 (0 programs) and have replaced the files in the fix stated also

### ISSUE THREAD

**Issue by Slee42069** [iptv-org/epg#2319]
### Describe your issue

hi guys, im running the command for tvtv.us and getting 403 returned, i replaced the files that were updated to fix this also ran git pull and install to make sure everythings up to date, but all the other epgs im using are fine, wondering if anyone can help at all ? :D 

or if not recommend a source to use for USA epg

root@Ubuntu-2004-focal-64-minimal-hwe ~/epg # npm run grab -- --site=tvtv.us

> grab
> npx tsx scripts/commands/epg/grab.ts --site=tvtv.us

starting...
config:
  output: guide.xml
  maxConnections: 1
  gzip: false
  site: tvtv.us
loading channels...
  found 2299 channel(s)
run #1:
  [1/4598] tvtv.us (en) - 33691 - Jan 29, 2024 (0 programs)
    ERR: Request failed with status code 403
  [2/4598] tvtv.us (en) - 33691 - Jan 30, 2024 (0 programs)
    ERR: Request failed with status code 403
  [3/4598] tvtv.us (en) - 83157 - Jan 30, 2024 (0 programs)
    ERR: Request failed with status code 403
  [4/4598] tvtv.us (en) - 5StarMaxEast.us - Jan 30, 2024 (0 programs)
    ERR: Request failed with status code 403
  [5/4598] tvtv.us (en) - ABCNewsLive.us - Jan 30, 2024 (0 programs)
    ERR: Request failed with status code 403
  [6/4598] tvtv.us (en) - AdultSwimWest.us - Jan 30, 2024 (0 programs)

**Comment by Sicilykill** [comment]
403 (forbidden) response would mean you are getting blocked . Maybe you are getting blocked by Cloudflare? 
Are you even able to hit the tvtv.us site and load channels through a browser? 
Are you using a VPN that maybe is getting blocked?  
The site works fine for me even if I pull the latest. See example below:

Cloned repo
```
$ git clone https://github.com/iptv-org/epg.git epg_clone_main
Cloning into 'epg_clone_main'...
remote: Enumerating objects: 44254, done.
remote: Counting objects: 100% (1790/1790), done.
remote: Compressing objects: 100% (212/212), done.
remote: Total 44254 (delta 1604), reused 1719 (delta 1578), pack-reused 42464
Receiving objects: 100% (44254/44254), 41.11 MiB | 11.81 MiB/s, done.
Resolving deltas: 100% (30165/30165), done.
```

Installed dependencies
```

$ cd epg_clone_main/
$ npm install

> postinstall
> npm run api:load


> api:load
> npx tsx scripts/commands/api/load.ts

channels.json      [====================================] 100.00% | ETA: 0s | 1.4 MB | N/A
regions.json       [====================================] 100.00% | ETA: 0s | 2.2 KB | N/A
countries.json     [====================================] 100.00% | ETA: 0s | 4.3 KB | N/A
subdivisions.json  [====================================] 100.00% | ETA: 0s | 36.6 KB | N/A

added 754 packages, and audited 755 packages in 22s

116 packages are looking for funding
  run `npm fund` for details

1 moderate severity vulnerability

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
```

Ran the tvtv.us grab command

```
$ npm run grab -- --site=tvtv.us

> grab
> npx tsx scripts/commands/epg/grab.ts --site=tvtv.us

starting...
config:
  output: guide.xml
  maxConnections: 1
  gzip: false
  site: tvtv.us
loading channels...
  found 2299 channel(s)
run #1:
  [1/4598] tvtv.us (en) - 33691 - Jan 29, 2024 (22 programs)
  [2/4598] tvtv.us (en) - 33691 - Jan 30, 2024 (25 programs)
  [3/4598] tvtv.us (en) - 83157 - Jan 30, 2024 (47 programs)
  [4/4598] tvtv.us (en) - 83157 - Jan 29, 2024 (47 programs)
  [5/4598] tvtv.us (en) - 5StarMaxEast.us - Jan 30, 2024 (15 programs)
  [6/4598] tvtv.us (en) - ABCNewsLive.us - Jan 30, 2024 (36 programs)
  [7/4598] tvtv.us (en) - AdultSwimWest.us - Jan 30, 2024 (27 programs)
  [8/4598] tvtv.us (en) - AnimalPlanetWest.us - Jan 30, 2024 (23 programs)

```

**Comment by Slee42069** [comment]
ahh okay so its being blocked then,thats perfect thankyou :)

im running it on a server hosted in Germany so ill probably just need to obscure that in that case, there isnt a built in proxy function too the epg puller all is there ? ill look at sorting it now, thankyou again :)

**Comment by chinochao** [comment]
I am also running the epg from a Github Action and it seems to be 403 error only in tvtv.us so far.

**Comment by Slee42069** [comment]
> I am also running the epg from a Github Action and it seems to be 403 error only in tvtv.us so far.

that's definitely you being blocked mate like @Sicilykill said above to me :) ive ran it from a different ip and its running fine :) 

just got to work out how best to change the ip of the server being blocked myself now :P

**Comment by Sicilykill** [comment]
yeah as @Slee42069 mentioned there is not much we can do to fix the 403. 
You can try finding a VPN that works.  That is about it. If you look at the request in your browser when accessing tvtv.us you will see they are behind cloudflare.  We are at their mercy and at any time could add add more rules to block incoming connections.
